File: backend/app/schemas/Garden.py
from pydantic import BaseModel, Field
from typing import Annotated, List
from app.schemas.User import UserPublic
from app.schemas.GardenPlant import GardenPlant
from app.schemas.GardenSection import SectionOut
from app.schemas.Image import ImageOut

class GardenCreate(BaseModel):
    name: Annotated[str, Field(max_length=30)]
    description: Annotated[str, Field(max_length=256)]
    is_public: bool
    tags: Annotated[list[int], Field(max_length=5)]


class GardenOut(BaseModel):
    id: int
    name: str
    description: str
    is_public: bool
    tags: List[Tag]
    user: UserPublic
    sections: List[SectionOut]
    garden_plants: List[GardenPlant]
    garden_images: List[ImageOut]


class Tag(BaseModel):
    id: int
    name: str

File: frontend/src/types/garden.ts
import type { UserPublic } from "./user";

export interface Garden {
    id: number;
    name: string;
    description: string;
    is_public: number;
    tags: Tag[];
    user_id: number;
    user: UserPublic;
    sections: Section[];
    garden_images: Image[];
}

export interface Tag {
    id: number;
    name: string;
}

export interface Section {
    id: number;
    name: string;
    description: string;
    order: number;
}

export type GardenPlantAmount = {
    plant_id: number,
    amount: number
}

export type Image = {
    id: number;
}

export type gardenImage = {
    gardenId: number;
    image: string | null;
}